modules/standard/Memory/Diagnostics.chpl

```diff
@@ -0,0 +1,251 @@
+    leaked memory by invoking :proc:`printMemAllocsByType` implicitly
+    when the program terminates normally. If there are no leaks,
+    nothing is printed.
+
+  ``memLeaksLog``: `string`:
+    Enable memory tracking and append a report to the named file when
+    the program terminates normally.
+    The report contains the compilation command used to build the
+    program, the execution command used to run it, the summary
+    allocation statistics produced by :proc:`printMemAllocStats`, and the
+    summary information about leaked memory produced by
+    :proc:`printMemAllocsByType`.
+    
+
+  ``memStats``: `bool`:
+    Enable memory tracking and produce summary memory statistics by
+    invoking :proc:`printMemAllocStats` implicitly at normal program
+    termination.  Note that for a multi-locale run, each top-level
+    locale reports its own memory statistics and these reports may
+    appear in any order in the program output (and possibly even
+    interleaved, though we have gone to some effort to avoid that).
+    These statistics can be put in order by sorting the output lines
+    that begin with the string ``memStats:``.
+
+  ``memMax``: `uint`:
+    If the value is greater than 0 (zero), enable memory tracking
+    and use the value as a limit on the number of bytes of memory
+    that can be allocated per top-level (network-connected) locale.
+    If during execution the amount of allocated memory exceeds this
+    limit on any locale, halt the program with a message saying so.
+
+  The following two config variables do not enable memory tracking;
+  they only modify how it is done.
+
+
+  ``memThreshold``: `uint`:
+    If this is set to a value greater than 0 (zero), only allocation
+    requests larger than this are tracked and/or reported.
+
+  ``memLog``: `string`:
+    Memory reporting is written to this file.  By default it is the
+    ``stdout`` associated with the process (not the Chapel channel
+    with the same name).  Setting this config variable to a file path
+    causes the reporting to be written to that named file instead.
+    In multilocale executions each top-level locale produces output
+    to its own file, with a dot ('.') and the locale ID appended to
+    this path.
+ */
+module Diagnostics {
+
+pragma "insert line file info"
+private extern proc chpl_memoryUsed(): uint(64);
+
+/*
+  The amount of memory returned by :proc:`locale.physicalMemory` can
+  be expressed either as individual bytes or as chunks of 2**10,
+  2**20, or 2**30 bytes.
+ */
+enum MemUnits {Bytes, KB, MB, GB};
+
+/*
+  How much physical memory is present on this locale?
+
+  Note that this is a physical quantity and does not take into account
+  things like virtual memory support which might allow for allocating
+  a larger amount, or system or user limits which might prevent
+  allocating so much.
+
+  *Note:* Unlike the other procedures in the :mod:`Diagnostics` module, this
```

Comment:
Should this and/or the previous paragraph be a Sphinx style note?

Reply:
Done and done, thanks!